applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_alm_variables_calculation_process.cpp

```diff
@@ -0,0 +1,133 @@
+
+            std::vector<NodeType::Pointer> element_nodes_2 (3);
+            element_nodes_2[0] = p_node_2;
+            element_nodes_2[1] = p_node_5;
+            element_nodes_2[2] = p_node_3;
+            Triangle3D3 <NodeType> triangle_2( PointerVector<NodeType>{element_nodes_2} );
+
+            std::vector<NodeType::Pointer> element_nodes_3 (3);
+            element_nodes_3[0] = p_node_5;
+            element_nodes_3[1] = p_node_6;
+            element_nodes_3[2] = p_node_3;
+            Triangle3D3 <NodeType> triangle_3( PointerVector<NodeType>{element_nodes_3} );
+
+            Condition::Pointer p_cond_0 = ThisModelPart.CreateNewCondition(rConditionName, 1, triangle_0, p_cond_prop);
+            Condition::Pointer p_cond_1 = ThisModelPart.CreateNewCondition(rConditionName, 2, triangle_1, p_cond_prop);
+            Condition::Pointer p_cond_2 = ThisModelPart.CreateNewCondition(rConditionName, 3, triangle_2, p_cond_prop);
+            Condition::Pointer p_cond_3 = ThisModelPart.CreateNewCondition(rConditionName, 4, triangle_3, p_cond_prop);
+        }
+
+        /**
+        * Checks the correct work of the ALM variables process
+        * Test triangle
+        */
+        KRATOS_TEST_CASE_IN_SUITE(ALMVariablesProcess, KratosContactStructuralMechanicsFastSuite)
+        {
+            Model this_model;
+            ModelPart& this_model_part = this_model.CreateModelPart("Main", 2);
+
+            this_model_part.AddNodalSolutionStepVariable(NODAL_H);
+
+            auto& r_process_info = this_model_part.GetProcessInfo();
+            r_process_info.SetValue(DOMAIN_SIZE, 3);
+            r_process_info.SetValue(STEP, 1);
+            r_process_info.SetValue(NL_ITERATION_NUMBER, 1);
+
+            Create3DConditionsGeometry(this_model_part, "SurfaceCondition3D3N");
+
+            // Assign NodalH and flags
+            for (auto& r_node : this_model_part.Nodes()) {
+                r_node.FastGetSolutionStepValue(NODAL_H) = static_cast<double>(r_node.Id()); // Adding some randomness
+//                 r_node.FastGetSolutionStepValue(NODAL_H) = 1.0;
+                r_node.Set(SLAVE);
+                r_node.Set(MASTER);
+            }
+            for (auto& r_cond : this_model_part.Conditions()) {
+                r_cond.Set(SLAVE);
+                r_cond.Set(MASTER);
+            }
+
+            // Compute ALM variables
+            ALMVariablesCalculationProcess alm_process(this_model_part);
+            alm_process.Execute();
+
+            const double tolerance = 1.0e-4;
+            const double initial_penalty = this_model_part.GetProcessInfo().GetValue(INITIAL_PENALTY);
+            const double scale_factor = this_model_part.GetProcessInfo().GetValue(SCALE_FACTOR);
+
+//             // DEBUG
+//             GiDIODebugALMVariables(this_model_part);
+
+            KRATOS_CHECK_LESS_EQUAL(std::abs((initial_penalty - 305.858)/initial_penalty), tolerance);
```

Comment:
why less equal and not equal?
I noticed you are using this often

Reply:
Because it is an absolute value

Reply:
ah sure